File: server/router/yunguan/res/resDevice.go
package res

import (
	"github.com/gin-gonic/gin"
	v1 "ygpt/server/api/v1"
)

type ResDeviceRouter struct {
}

func (s *ResDeviceRouter) InitResDeviceRouter(Router *gin.RouterGroup) {
	dataRouter := Router.Group("res/device")
	diskApi := v1.ApiGroupApp.YunGuanApiGroup.ResApi.ResDeviceApi

	{
		dataRouter.POST("addDevice", diskApi.AddResDevice)                 // 添加数据
		dataRouter.PUT("updateDevice", diskApi.UpdateResDevice)            // 修改数据
		dataRouter.DELETE("deleteDevice", diskApi.DeleteResDevice)         // 删除数据
		dataRouter.POST("queryDevice", diskApi.QueryResDevice)             // 获取数据列表
		dataRouter.POST("queryAllServer", diskApi.QueryAllServerResDevice) // 获取数据列表
	}
}

File: web/src/api/yunguan/resource/resDevice.js
import request from '@/utils/request'

// 添加元素
export function addDevice(modelData) {
    return request({
        url: 'res/device/addDevice',
        method: 'post',
        data: modelData
    })
}

// 修改元素
export function updateDevice(params) {
    return request({
        url: 'res/device/updateDevice',
        method: 'put',
        data: params
    })
}

// 获取节点元素列表
export function getDeviceList(params) {
    return request({
        url: 'res/device/queryDevice',
        method: "post",
        data:params
    })
}
// 获取节点元素列表
export function getAllServerList() {
    return request({
        url: 'res/device/queryAllServer',
        method: "post",
    })
}

// 删除元素
export function deleteDevice(params) {
    return request({
        url: 'res/device/deleteDevice',
        method: "delete",
        data: {
            id: params
        }
    })
}
